doc/src/manual/functions.md

```diff
@@ -711,41 +711,58 @@ call will fail, just as it would if too many arguments were given explicitly.
 
 It is often possible to provide sensible default values for function arguments.
 This can save users from having to pass every argument on every call.
-For example, the function [`Date(y, [m, d])`](@ref)
-from `Dates` module constructs a `Date` type for a given year `y`, month `m` and day `d`.
-However, `m` and `d` arguments are optional and their default value is `1`.
-This behavior can be expressed concisely as:
-
-```julia
-function Date(y::Int64, m::Int64=1, d::Int64=1)
-    err = validargs(Date, y, m, d)
-    err === nothing || throw(err)
-    return Date(UTD(totaldays(y, m, d)))
-end
+For example, suppose we want to create a simple function to take in integers and then return a
+string of those integers in a date format; instead of having the user pass the month and day always,
+we can make them optional by giving default values to them as `1`. This behavior can be expressed
+concisely as:
+
+```jldoctest datefunc
+julia> function date(year::Int, month::Int=1, day::Int=1)
+           y = clamp(year,  1:typemax(Int))
+           m = clamp(month, 1:12)
+           d = clamp(day,   1:31)
+           if m == 2
+               is_leap_year = ((y%4 == 0) && (y%100 ≠ 0)) || (y%400 == 0)
+               is_leap_year ? d = clamp(d, 1:29) : d = clamp(d, 1:28)
+           elseif m in [4, 6, 9, 11]
+               d = clamp(d, 1:30)
+           end
+           yy = string(y, pad=4)
+           mm = string(m, pad=2)
+           dd = string(d, pad=2)
+           return "$yy-$mm-$dd"
+       end
+date (generic function with 3 methods)
 ```
 
-Observe, that this definition calls another method of the `Date` function that takes one argument
-of type `UTInstant{Day}`.
+Note how the signature says **"date (generic function with 3 methods)"**, denoting that
+we have a function named `date`, and there are 3 "versions" (i.e. methods) of it.
```

Comment:
This is redundant with lines 757-765.

Reply:
I thought it would be a surprise to see that as `3`, so it was necessary to give a few words on that. Anyways, I see it's redundant now, since users will definitely see it at the bottom.